DRAFT blogpost core retreat (#2289)
nf-core/website: src/content/blog/2024/core_retreat_Feb_2024.mdx

---
title: 'core team retreat'
subtitle: 'funny Viking themed title here'
pubDate: 2024-01-29T00:00:00+01:00
headerImage: '/assets/images/blog/core_retreat_Feb_2024/group_pic4.jpg'
headerImageAlt: 'Viking core team'
authors:
    - 'FranBonath'
label:
    - 'community'
embedHeaderImage: false
---

import work_pic1 from '@assets/images/blog/core_retreat_Feb_2024/work_pic1.png';
import viking1 from '@assets/images/blog/core_retreat_Feb_2024/viking1.jpg';
import viking2 from '@assets/images/blog/core_retreat_Feb_2024/viking2.jpg';
import viking3 from '@assets/images/blog/core_retreat_Feb_2024/viking3.jpg';
import viking4 from '@assets/images/blog/core_retreat_Feb_2024/viking4.jpg';
import viking5 from '@assets/images/blog/core_retreat_Feb_2024/viking5.jpg';
import viking6 from '@assets/images/blog/core_retreat_Feb_2024/viking6.jpg';
import viking7 from '@assets/images/blog/core_retreat_Feb_2024/viking7.jpg';
import viking8 from '@assets/images/blog/core_retreat_Feb_2024/viking8.jpg';
import fin from '@assets/images/blog/core_retreat_Feb_2024/fin.jpg';

import { Image } from 'astro:assets';

<Image src={work_pic1} alt="core team working at retreat" />

> Between the 5th and 8th of February, the core team and members of the Steering group met in icy Stockholm to discuss upcoming challenges and improvements to the community. The topics and desicions are outlined below.

Reviewer: ```suggestion
Between the 5th and 8th of February, the core team and members of the steering group met in icy Stockholm to discuss upcoming challenges and improvements to the community. The topics and decisions are outlined below.
```

# Structure and Governance

-   **Funding**

>     As many open-source communities, funding is an important topic. The nf-core community has applied for another grant from the Chan Zuckerberg Initiative, but we will not hear about a decision until later in the year. Until then, the community is run on volunteer work by all of you and supported by different Universities and institutions.

Reviewer: ```suggestion
    As many open-source communities, funding is an important topic. The nf-core community has applied for another grant from the Chan Zuckerberg Initiative, but we will not hear about a decision until later in the year. Until then, the community is run on volunteer work by all of you and supported by different universities and institutions.
```

> -   **nf-core trademark**
> 
>     Currently, the nf-core logo is not trademarked. During the retreat the core team and steering group discussed the benefits and drawbacks of trademarking the nf-core logo.
> 

Reviewer: ```suggestion
```

> -   **nf-core working groups**
> 
>     An exiting development is that nf-core wants to establish working groups / SIGs / IGs / COSIs to encourage people within the nf-core community with similar interest to meet and work together. More information on this new initiative will be presented in the bytesize talk on February 20th [link to the bytesize page].

Reviewer: ```suggestion
-   **nf-core communities**

    An exiting development is that nf-core wants to establish research communities to encourage people within the nf-core community with similar interest to meet and work together. More information on this new initiative will be presented in the [bytesize talk on February 20th](/events/2024/bytesize_workgroups_intro).
```

# Outreach and Community

-   **Training and Documentation overhaul**

>     One of the main goals of nf-core is to provide easy to use and well documented tools and pipelines. While the pipeline documentations are in a good state, the docs for tools and training are a bit scattered throughout the website and at times not very accessible. You can look forward to a complete overhaul of the docs page towards a more modular setup, which will also help putting together more targeted training for different levels of knowledge. Be aware that this is a major project and will likely require a good amount of time.

Reviewer: ```suggestion
    One of the main goals of nf-core is to provide easy to use and well documented tools and pipelines. While the pipeline documentation pages are generally in a good state, the docs for tools and training are a bit scattered throughout the website and at times not very accessible. You can look forward to a complete overhaul of the docs page towards a more modular setup, which will also help putting together more targeted training for different levels of knowledge. Be aware that this is a major project and will likely require a good amount of time.
```

-   **Publication of a nf-core v2.0 paper**

>     It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed masively since then. Not only has it tremendously grown in size, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community build around FAANG that sparked the idea about creating work groups.

Reviewer 2: ```suggestion
    It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed massively since then. Not only has it tremendously grown in size, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community build around FAANG that sparked the idea about creating work groups.
```

>     It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed masively since then. Not only has it tremendously grown in size, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community build around FAANG that sparked the idea about creating work groups.

Reviewer 2: ```suggestion
    It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed masively since then. Not only has it tremendously grown in size, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community built around FAANG that sparked the idea about creating work groups.
```

>     It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed masively since then. Not only has it tremendously grown in size, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community build around FAANG that sparked the idea about creating work groups.

Reviewer: > [insert gif or something]

Reviewer: ```suggestion
    It has been 4 years since the first nf-core paper was published [insert gif or something] and the community has changed a lot since then. Not only has it grown tremendously, but we also experienced a lot of technical changes including the update to Nextflow DSL2, the introduction of modules and subworkflows and upgrades to nf-core/tools. To reflect these changes, the core team would like to publish an updated state of nf-core paper. A first draft is in the works and will also emphasize the work done in the community build around FAANG that sparked the idea about creating nf-core communities.
```

> -   **Crediting nf-core components in non-nf-core pipelines**
> 
>     Many non-nf-core pipelines use the nf-core tooling or nf-core modules and some specifically want to follow the nf-core guidelines while not being an official nf-core pipeline. It was dicussed whether or not we want to give a platform for such pipelines to be highlighted on the website.
> 

Reviewer: ```suggestion
```

-   **nf-core mentorship program**

    The three previous nf-core mentorship rounds have been a great success. However, with the uncertainty of further funding, it was a major discussion point how nf-core would be able to provide continues opportunities for both new, but also more experienced members of the communities. There are plans to change the mentorship program to be volunteer based, whereas prospective mentors can set more targeted aims and community members can apply to me a mentee of (a) specific mentor(s). More information will be available soon, so keep your eyes on the announcement channel in Slack.

-   **Open office hours for nf-core developers**

>     As a trial, nf-core wants to implement weekly open office hours to help with reviews and discuss issues developers experience during their work on nf-core related items. Be aware that this explicitly **won't** be a training event. Stay tuned for more information.

Reviewer: ```suggestion
    As a trial, nf-core wants to implement weekly open office hours to bring people together to collaborate on pull-request reviews and discuss issues developers experience during their work on nf-core related items. Be aware that this explicitly **won't** be a training event. Stay tuned for more information.
```

# Technical Improvements

-   **Roadmap for nf-core/tools**
    

>     There are lots of smaller and bigger improvements to nf-core/tools in the works, but special mention deserves the config builder that will aid in the currently somewhat cumbersom creation of institutional or other config files to run nf-core and other nextflow pipelines.

Reviewer: ```suggestion
    There are lots of smaller and bigger improvements to nf-core/tools in the works, but special mention deserves the config builder that will help new users to create institutional or other config files to run nf-core and other nextflow pipelines.
```

-   **Restructuring of nf-test and nf-core/modules**

-   **Reference genomes**
    

>     The core team started to collect ideas on how to improve iGenomes.

Reviewer: ```suggestion
    The core team started to collect ideas on how to improve AWS-iGenomes.
```

-   **Resource optimization**

>     The goal is to improve resource usage on the module level and as a first step it is planned to make use of optimized configs taken from Seqera platform®.

Reviewer: ```suggestion
    The goal is to improve resource usage on the module level and as a first step it is planned to make use of optimized configs taken from Seqera Platform®.
```

-   **Benchmarking**

    Wouldn't it be great to know how a nf-core pipeline compares to other pipelines or to a previous version? The core team decided to do a proof of concept for nf-core/sarek and nf-core/multipesequencealign, to investigate the feasibility and time demands.

-   **Wave and nf-core**

>     Seqera wave® is a way to generate containers "on the fly" and it was discussed if and how wave® could be adopted to be used in nf-core.

Reviewer: ```suggestion
    [Seqera Wave](https://seqera.io/wave/) is a way to generate containers "on the fly" and it was discussed if and how Wave could be adopted to be used in nf-core.
```

---
title: 'core team retreat'

> subtitle: 'funny Viking themed title here'

Reviewer 3: ```suggestion
subtitle: 'Why not come to Stockholm in February?🥶'
```

Reviewer: Needs a space before the emoji 😉 

```suggestion
subtitle: 'Why not come to Stockholm in February? 🥶'
```

### Benchmarking

Wouldn't it be great to know how a nf-core pipeline compares to other pipelines or to a previous version?
The core team decided to do a proof of concept for `nf-core/sarek` and `nf-core/multipesequencealign`, to investigate the feasibility and time demands.

### Wave and nf-core

[Seqera Wave](https://seqera.io/wave/) is a way to generate software containers and it was discussed if and how Wave could be used to help developers in nf-core.

# nf-core/vikings

As is becoming tradition for nf-core events in Stockholm, the group visited at the viking-themed restaurant [Aifur](https://aifur.se/).
Much silliness ensued.

<Image src={viking1} alt="first viking" height="300" />
<Image src={viking2} alt="second viking" height="300" />
<Image src={viking3} alt="third viking" height="300" />
<Image src={viking5} alt="fourth viking" height="300" />
<Image src={viking6} alt="fifth viking" height="300" />
<Image src={viking7} alt="sixth viking" height="300" />
<Image src={viking8} alt="seventh viking" height="300" />
<Image src={viking4} alt="eighth viking" height="300" />

# fin

> All the excitement was too much for some of the team members..

Reviewer 4: ```suggestion
All the excitement was too much for some of the team members.
```
